src/exe/cimbar_recv/recv.cpp

```diff
@@ -130,29 +136,41 @@ int main(int argc, char** argv)
 			continue;
 		}
 
-		cv::UMat img = mat.getUMat(cv::ACCESS_RW).clone();
+		cv::Mat img = mat.clone();
 		cv::cvtColor(mat, mat, cv::COLOR_BGR2RGB);
 
 		// draw some stats on mat?
 		window.show(mat, 0);
 
-		// extract
-		bool shouldPreprocess = false;
-		int res = ext.extract(img, img);
-		if (!res)
+		// extract, decode, etc
+		int bytes = cimbard_scan_extract_decode(img.data, img.cols, img.rows, 3, bufspace.data(), bufspace.size());
+		if (bytes <= 0)
+			continue;
+
+		if (bytes % chunkSize != 0)
 		{
-			//std::cerr << "no extract " << mat.cols << "," << mat.rows << std::endl;
+			std::cerr << "WEIRD SEEMS BAD? " << count << std::endl;
```

Comment:
I don't think this is actually possible. We'll remove it when we have more confidence.